model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -207,22 +207,30 @@ message PCollection {
 // ProcessBundleDescriptor.
 message ExecutableStagePayload {
 
-  // Environment in which this stage executes. We use an environment rather than environment id
+  // (Required) Environment in which this stage executes.
+  //
+  // We use an environment rather than environment id
   // because ExecutableStages use environments directly. This may change in the future.
   Environment environment = 1;
 
-  // Input PCollection id.
+  // (Required) Input PCollection id. This must be present as a value in the inputs of any
+  // PTransform the ExecutableStagePayload is the payload of.
   string input = 2;
 
-  // Side Input PCollection ids.
+  // (Required) Side Input PCollection ids. Each must be present as a value in the inputs of
```

Comment:
It's weird to indicate that a repeated field is "required" since the empty list will always be accepted. It seems sufficient to claim that each input must be specified here.

Reply:
removed, added a note to `transforms` that it must contain at least one transform.